crates/shared/src/oneinch_api.rs

```diff
@@ -384,6 +386,60 @@ where
     serde_json::from_str(&response?).context("1inch result parsing failed")
 }
 
+#[derive(Debug, Clone)]
+pub struct ProtocolCache(Arc<Mutex<TimedSizedCache<(), Vec<String>>>>);
```

Comment:
Can this just be `TimedCache` since the size can only ever be 1 so we don't care about it?

Reply:
I think that is correct.

Reply:
They should behave the same in our case but their implementation is different. `SizedCache` and `TimedSizedCache` use quite a bit of `unsafe` in the authors homegrown implementation of the inner map storing the cached values.
`TimedCache` uses a simple `std::collections::HashMap` so let's go with that instead.